lib/chef/knife/oci_server_create.rb

```diff
@@ -212,7 +227,44 @@ def wait_for_ssh(hostname, ssh_port, interval_seconds, max_time_seconds)
         false
       end
 
-      def can_ssh(hostname, ssh_port)
+      # returns a Net::SSH:Gateway object or nil
+      def get_ssh_gateway(ssh_gateway)
+        # Probably should be doing a more complete lookup here.
+        # For example, looking in ssh configuration files, etc.
+        return nil unless ssh_gateway
+        gw_host, gw_user = ssh_gateway.split('@').reverse
+        gw_host, gw_port = gw_host.split(':')
+        gateway_options = { port: gw_port || SSH_PORT }
+        ssh_gateway_config = Net::SSH::Config.for(gw_host)
+        gw_user ||= ssh_gateway_config[:user]
+
+        # stash a copy of the keys, temporarily
+        gateway_keys = ssh_gateway_config[:keys]
+        # override with user specified keys
+        if config[:ssh_gateway_identity]
```

Comment:
Is this option exposed to users? I'm not seeing it.

Reply:
This will be used when '--ssh-gateway username@hostname:port' is given by the user.  (Or the short form of '-G username@hostname:port')

Reply:
But where does config[:ssh_gateway_identity] get set?